modules/acid/src/main/ts/ephox/acid/gui/ColourPicker.ts

```diff
@@ -61,9 +67,28 @@ const makeFactory = (
       });
     };
 
-    const runUpdates = (anyInSystem: AlloyComponent, hex: Hex, updates: ((anyInSystem: AlloyComponent, hex: Hex) => void)[]) => {
+    const updateSlider = (anyInSystem: AlloyComponent, _hex: Hex, hue: number) => {
+      memSlider.getOpt(anyInSystem).each((slider) => {
+        Slider.setValue(slider, { y: hueDegreesToSlider(hue) });
+      });
+    };
+
+    const updatePaletteThumb = (anyInSystem: AlloyComponent, hex: Hex) => {
+      memPalette.getOpt(anyInSystem).each((palette) => {
+        sbPalette.setThumb(palette, hex);
+      });
+    };
+
+    const updateState = (hex: Hex, hue: number) => {
+      const rgba = RgbaColour.fromHex(hex);
+      state.paletteRgba.set(rgba);
+      state.paletteHue.set(hue);
+    };
+
+    const runUpdates = (anyInSystem: AlloyComponent, hex: Hex, hue: number, updates: ((anyInSystem: AlloyComponent, hex: Hex, hue: number) => void)[]) => {
+      updateState(hex, hue);
```

Comment:
This doesn't quite look right, as it means each update won't be able to get the previous state?

Reply:
`runUpdates` runs after the old state has been retrieved by the update functions (i.e. `runUpdates` is called with all the new values, including the ones derived from the old state).

The order of operations in here shouldn't matter, but we can move it if it's confusing?

Reply:
Discussed on a call and I got confused by the `updatePalette` vs `paletteUpdates`. So we're going to rename the event handlers to start with `on` to try and avoid that confusion.

Reply:
This got us both within 48 hours, so they've been renamed: https://github.com/tinymce/tinymce/pull/6526/files#diff-9949b93a6a2a458146049c9e5af75971f5e6a54a06d554531f4a5412bdd18258R94